File: test/langs/sample.c
#include <stdio.h>

struct Ledger {
    long balance;
};

long deposit(struct Ledger *ledger, long amount) {
    ledger->balance += amount;
    return ledger->balance;
}

int main(void) {
    struct Ledger ledger = {0};
    deposit(&ledger, 100);
    return 0;
}

File: test/langs/sample.rs
struct Ledger {
    balance: i64,
}

enum Currency {
    Usd,
    Eur,
}

trait Account {
    fn balance(&self) -> i64;
}

impl Ledger {
    fn deposit(&mut self, amount: i64) {
        self.balance += amount;
    }
}

fn new_ledger() -> Ledger {
    Ledger { balance: 0 }
}
